aptos-move/framework/src/natives/string_utils.rs

```diff
@@ -312,23 +313,46 @@ fn native_format_impl(
                 && type_.module.as_str() == "option"
                 && type_.address == AccountAddress::ONE
             {
-                let mut v = strct
-                    .unpack()?
-                    .next()
-                    .unwrap()
-                    .value_as::<Vector>()?
-                    .unpack_unchecked()?;
-                if v.is_empty() {
-                    out.push_str("None");
+                if context
```

Comment:
Can you add comments for what you are doing and perhaps pull in a different function?

Reply:
done